incubator/virtualcluster/cmd/kubectl-vc/use.go

```diff
@@ -0,0 +1,132 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/multi-tenancy/incubator/virtualcluster/pkg/syncer/conversion"
+)
+
+const vcKubeconfigHome = ".kube/vc/"
+
+var useCommand = cli.Command{
+	Name:      "use",
+	Usage:     "Switch kubeconfig among virtualclusters",
+	ArgsUsage: "VC_NAME",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "namespace, n",
+			Usage: "If present, the namespace scope for this CLI request, or use 'default'",
+		},
+		cli.BoolFlag{
+			Name:  "interactive, i",
+			Usage: "Launch a shell and switch to the specific virtualcluster kubeconfig",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		ns := metav1.NamespaceDefault
+		if fns := cctx.String("namespace"); fns != "" {
+			ns = fns
+		}
+
+		name := cctx.Args().Get(0)
+		if name == "" {
+			return errors.New("virtualcluster name is required")
+		}
+
+		kbFilePath, err := placeVCKubeconfig(ns, name)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("kubeconfig for virtualcluster %s/%s is placed at:\n\n\t%s\n\n", ns, name, kbFilePath)
+
+		if cctx.Bool("interactive") {
+			return enterVCShell(kbFilePath, ns, name)
+		}
+
+		return nil
+	},
+}
+
+func placeVCKubeconfig(ns, name string) (string, error) {
+	vccli, err := newVCClient()
+	if err != nil {
+		return "", errors.Wrapf(err, "new vc client")
+	}
+
+	vc, err := vccli.TenancyV1alpha1().VirtualClusters(ns).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	cv, err := vccli.TenancyV1alpha1().ClusterVersions().Get(vc.Spec.ClusterVersionName, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrapf(err, "cluster version not found")
+	}
+
+	cli, err := newGenericK8sClient()
+	if err != nil {
+		return "", err
+	}
+
+	kbBytes, err := genKubeConfig(cli, vc, cv)
+	if err != nil {
+		return "", err
+	}
+
+	kbFileDir := filepath.Join(os.Getenv("HOME"), vcKubeconfigHome)
+	if err = os.MkdirAll(kbFileDir, 0755); err != nil {
+		return "", err
+	}
+
+	kbFilePath := filepath.Join(kbFileDir, conversion.ToClusterKey(vc)+".kubeconfig")
+	err = ioutil.WriteFile(kbFilePath, kbBytes, 0644)
+
+	return kbFilePath, err
+}
+
+func enterVCShell(kbFilePath, ns, name string) error {
+	fmt.Printf("❗ You are now at VirtualCluster %s/%s\n", ns, name)
+	fmt.Println("❗ type 'exit' to exit this temporary workspace")
+
+	bashBinPath, err := exec.LookPath("bash")
```

Comment:
Can we use $SHELL instead of bash? somthing like, exec.LookPath(os.Getenv("SHELL"))

Reply:
good idea